alt-tab-macos/logic/Screen.swift

```diff
@@ -14,9 +14,23 @@ class Screen {
 
     static func updateThumbnailMaxSize() -> Void {
         if Preferences.thumbnailMaxWidth == 200 && Preferences.thumbnailMaxHeight == 200 {
-            let main = NSScreen.main!.frame
+            let screenFrame = getPreferredScreen().frame
             Preferences.thumbnailMaxWidth = (NSScreen.main!.frame.size.width * Preferences.maxScreenUsage! - Preferences.windowPadding * 2) / Preferences.maxThumbnailsPerRow! - Preferences.interItemPadding
-            Preferences.thumbnailMaxHeight = Preferences.thumbnailMaxWidth * (main.height / main.width)
+            Preferences.thumbnailMaxHeight = Preferences.thumbnailMaxWidth * (screenFrame.height / screenFrame.width)
         }
     }
```

Comment:
I think there is a bug here. This method is called when `NSApplication.didChangeScreenParametersNotification` happens. The use-case is the user changes their display, so we should recalculate the size of thumbnails based on `Preferences.maxThumbnailsPerRow`. Already when I decided to remove the thumbnails size from the Preferences panel and make is a derived value from `maxThumbnailsPerRow`, I thought I may want to remove the thumbnail width/height from the `Preferences`. I didn't though.

Now that you add this PR, it becomes clearer that the width/height of thumbnail is a dynamic value and not at all a preferences.

I would suggest these changes:

* Remove `Preferences.thumbnailMaxWidth` and `Preferences.thumbnailMaxHeight` from being `Preferences`
* Rename `updateThumbnailMaxSize` and make it return `(width, height)` instead of storing in `Preferences`
* Call `updateThumbnailMaxSize` in place which use `Preferences.`width/height today: `layout.estimatedItemSize`, `layoutAttributesForElements` and `computeDownscaledSize`

Reply:
Independent of that I see Ive overlooked one usage of `NSScreen.main` here at line 18.

Reply:
My bad, this needs to be done in this commit/pull request or the feature will be released bugged. Gonna do the required changes here.

Reply:
Ok did a break but are now back on it. I profiled the application, it are indeed cheap calls. 

I refactored as you suggested with additional required changes:
We need to make sure the same NSScreen is used for all computations during one possibly long ongoing UI summon process. 

The call to `getPrefferedScreen()` was called many times during different computations at different places. I am not familiar enough with the call hierarchy yet and don't wanted to let them all to drag the `screenOverride` argument with them (through all method invocations inside the whole UI summon process) so I added `Screen.preferredScreenOverride` which  is set by `Application.showUiOrCycleSelection()` and unset by `Application.hideUi()` and then used by default by `getPreferredScreen()` as long as the value is not `nil` and the method caller does not say otherwise.

Tell me if its ok for you for now.

Reply:
But I must say I dont like that `preferredScreenOverride` as a resolution of this. 

It feels like the structure has to be reviewed and changed but I need more time for that to get familiar with all of your code for rendering. 

`Screen` class and all of its methods is static, so its more like a helper class. The current solution with `preferredScreenOverride` limits the application to one view at a time or lets say one screen. I think the callers of `Screen` (`calcThumbnailMaxSize`, `calcFrameMaxSize`, `showCenteredFrontPanel`) would need to provide which screen this methods should use, possibly as an optional argument.

Reply:
Sorry for another commit to that. Ive redone and tested it again in a dual monitor situation with both monitors having hugely different resolutions.

Corresponding part of commit message:
```
- Screen.preferredScreenOverride: removed
- instead currentScreen gets provided or updated now at all required places before triggering UI summon calculations
  - ThumbnailsPanel.computeThumbnails(): re-creates collectionViewLayout instead of invalidating it
```

Eventually we should reinstantiate ThumbnailsPanel on each AltTab invocation. I'm curious about your judgement.

Cheers.